### DISCUSSION TITLE
2.x problems

### DISCUSSION THREAD

**Discussion by King-Oni** [vhqtvn/VHSCode-Android#146] [General]
I really love this project but recently it got many updates that aren't very good.
Sarting from the UI changes that made it really hard to use to the rendering issues with the focusing
It looks like the new version tries to fix none existing issues

To me 1.19 is the best ever update made to this project I can see why the changes are made <maybe?>
But it hurts me to see the UI in my tablet the UI is hard and it makes me wants to get out of any work I do in it

On the other side I see 1.19 is better than 2.1 or any upcoming updates

**Comment by vhqtvn** [comment]
Thank you for your comment.

This project at first is made for my own use, and keep on adding features that I need. I decided to share with a hope that community can help me to have some actual application that can mobilize my workflow better. Until now some features fixed or modified thanks to community feedbacks and contributions.

At the early time I hoped to see VSCode on Android for portability to suite with my workflows that contains massive number of projects and development style of minor-edit-only most of the time, after failure of porting Electron to Android I found code-server and that lead to this project. Quite happy porting it until early version and found out limitations comparing to actual VSCode i decided to resolve the limitations with "Remote server" feature for remote accessing repos, "vsboot.js" feature for patching the editor without modifying the app, and then termux-mirroring to have basic env for setting up local scripting helpers, they defined the 1.x branch.

After a while I feel that with my workflow having 2 separating applications JuiceSSH and this editor for separate usage is not very good. So 2.x born, this editor become a workspace application for my target that is having something I can bring with my phones to remotely control servers, k8s clusters and minor modifications to various projects. Also 2.x try to make the editor to be more computer-like with virtual cursor (as tapping the screen is quite hard for some controls, and touch support is not very good too). My plan is to have an application that can manage ssh and has an editor to use when needed, but after some thoughts more general to ssh is a terminal connection, so I copied termux terminal into this editor as a base so i can think of a way to add things like ssh (i'm imagine it as a "context" within this application, we may haave more "contextes" other than ssh), but then i stuck on how to do the UX, how to implement these for easy usage and other people that already using the editor can understand. I can just add a ugly listbox for myself to suite my need this time but as this application get quite many people using it's hard for me to think a balance way and everythings result in current design - I'm bad at UX :(

To have it better I hope to get more feedbacks/ideas from other users and contributions if possible is very welcome. Until now github opened this discussion feature there's a chance for people to talk, as the version on google play is quite dead due to the new restriction. Most open issues til now is error from user, code server, termux, or limitations of these so i just leave them there. Another reason is that i do not have much time to work on this project frequently.

**Reply by King-Oni** [reply]
The cursor is actually a feature I personally like, and the 2.x update I thought it was came for the small device, to make it easy for them
But the UI/UX changes I'm happy to hear that they may change to better in the future

**Reply by vhqtvn** [reply]
Yes me also hope to see someone with good UX skills can help me draw some better screens, that will help me too as I'm not satisfied current screens also :( anw they works...

**Comment by vhqtvn** [comment]
For the rendering focus issue you can disable virtual mouse to get 1.x behavior, I do notice that but havent spent time investigating on what caused that (not an easy thing I guess), it's on my todo anyway.

Maybe current 2.x can change back to 1.x by adding a default option that will open the editor by default, that's what I intended to do at first but I dont know/want to let user read documents to use the new features, also i dont want to write documents that require screenshots that will need to be updated anytime I change the screens (again, i'm not good on UX so i just change according to my needs), so the guide screens came that slowdown the process to editor screen.

**Reply by King-Oni** [reply]
About that instead of screenshots you can make a little text box that points to settings and in settings a pointer with Box that say something
Like old design that way it's easy

**Reply by vhqtvn** [reply]
Hm ok added to todo.
